src/ripple/rpc/impl/RPCHelpers.cpp

```diff
@@ -671,44 +652,71 @@ keypairForSignature(Json::Value const& params, Json::Value& error)
         }
     }
 
-    // ripple-lib encodes seed used to generate an Ed25519 wallet in a
-    // non-standard way. While we never encode seeds that way, we try
-    // to detect such keys to avoid user confusion.
-    if (secretType != jss::seed_hex.c_str())
+    boost::optional<Seed> seed;
     {
-        seed = RPC::parseRippleLibSeed(params[secretType]);
+        // Check the key type and set the error if needed. returns true if no error,
+        // false if error
+        auto checkKeyType = [&error](boost::optional<KeyType> const& expected, boost::optional<KeyType> const& got) -> bool{
```

Comment:
Two comments:

1) Is a trailing return required here?
2) I'm wondering whether an almost 50-line lambda deserves to be a standlone function.

Reply:
Addressed in f69d726530ea5dc51951985db5b2a88ea8c9252e

BTW, it was only ~20 lines, not 50 :-)